src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -175,6 +184,75 @@ func (c *controller) WaitAndUpdateNodesStatus(status *ControllerStatus) {
 	c.log.Infof("Done waiting for all the nodes. Nodes in error status: %d\n", hostsInError)
 }
 
+func (c *controller) HackDNSAddressConflict(wg *sync.WaitGroup) {
+
+	c.log.Infof("Making sure service %s can reserve the .10 address", dnsServiceName)
+
+	defer wg.Done()
+	networks, err := c.kc.GetServiceNetworks()
+	if err != nil || len(networks) == 0 {
+		c.log.Errorf("Failed to get service networks: %s", err)
+		return
+	}
+
+	ip, _, _ := net.ParseCIDR(networks[0])
+	ip4 := ip.To4()
+	if ip4 == nil {
+		c.log.Infof("Service network is IPv6: %s, skipping the .10 address hack", ip)
+		return
+	}
+	ip4[3] = 10 // .10 is the conflicting address
+
+	for i := 0; i < maxDNSServiceIPAttempts; i++ {
+		svs, err := c.kc.ListServices("")
+		if err != nil {
+			c.log.WithError(err).Warnf("Failed to list running services, attempt %d/%d", i+1, maxDNSServiceIPAttempts)
+			time.Sleep(DNSAddressRetryInterval)
+			continue
+		}
+		s := c.findServiceByIP(ip4.String(), &svs.Items)
+		if s == nil {
+			c.log.Infof("No service found with IP %s, attempt %d/%d", ip4, i+1, maxDNSServiceIPAttempts)
+			time.Sleep(DNSAddressRetryInterval)
+			continue
+		}
+		if s.Name == dnsServiceName && s.Namespace == dnsServiceNamespace {
+			c.log.Infof("Service %s has successfully taken IP %s", dnsServiceName, ip4)
+			break
+		}
+		c.log.Warnf("Deleting service %s in namespace %s whose IP %s conflicts with %s", s.Name, s.Namespace, ip4, dnsServiceName)
+		if err := c.killConflictingService(s); err != nil {
+			c.log.WithError(err).Warnf("Failed to delete service %s in namespace %s", s.Name, s.Namespace)
+			break
+		}
+		if err := c.deleteDNSOperatorPods(); err != nil {
+			c.log.WithError(err).Warn("Failed to delete DNS operator pods")
+			break
```

Comment:
why to exit? maybe pod didn't start yet from previous restart? 
I think we should continue